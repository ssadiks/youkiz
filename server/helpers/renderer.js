import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import serialize from 'serialize-javascript';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Routes from '../../src/Routes';
import HeaderContainer from '../../src/components/container/HeaderContainer/HeaderContainer';

const renderer = (req, store) => {
  const muiTheme = getMuiTheme({
    userAgent: req.headers['user-agent']
  });

  const content = renderToString(
    <Provider store={store}>
      <MuiThemeProvider muiTheme={muiTheme}>
        <StaticRouter location={req.path} context={{}}>
          <div>
            {renderRoutes(Routes)}
          </div>
        </StaticRouter>
      </MuiThemeProvider>
    </Provider>
  );

  return `
    <html>
      <head>
        <link href="css/style.css" rel="stylesheet">
      </head>
      <body>
        <div id="root">${content}</div>
        <script>
          window.INITIAL_STATE = ${serialize(store.getState())}
        </script>
        <script src="js/vendors.js"></script>
        <script src="js/bundle.js"></script>
      </body>
    </html>
  `;
};

export default renderer;
