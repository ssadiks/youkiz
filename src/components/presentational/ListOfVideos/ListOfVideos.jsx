import React from 'react';
import PropTypes from 'prop-types';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import IconButton from 'material-ui/IconButton';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

const ListOfVideos = props => (
  <div className="ListOfVideos">
    <h2>{`${props.videosList.length} result(s)`}</h2>
    <div className="ListOfVideos__list">
      {
        props.videosList.map((video) => {
          const imageUrl = `https://i.ytimg.com/vi/${video.videoId}/mqdefault.jpg`;

          return (
            <div className="ListOfVideos__item" key={video.videoId}>
              <a
                className="ListOfVideos__item__link"
                href={`//www.youtube.com/watch?v=${video.videoId}`}
                target="_blank"
              >
                <img className="media-object" src={imageUrl} alt="" />
              </a>
              <div className="ListOfVideos__item__description">
                <span>{video.song}</span>
                <span>{video.type}</span>
                <IconMenu
                  iconButtonElement={<IconButton><MoreVertIcon /></IconButton>}
                  anchorOrigin={{horizontal: 'left', vertical: 'top'}}
                  targetOrigin={{horizontal: 'left', vertical: 'top'}}
                  className="ListOfVideos__item__more"
                >
                  <MenuItem primaryText="Edit" />
                  <MenuItem primaryText="Delete" onClick={() => props.deleteVideoAction(video._id)} />
                  <MenuItem primaryText="Infos" />
                </IconMenu>
              </div>
            </div>
          );
        })
      }
    </div>
  </div>
);

ListOfVideos.propTypes = {
  videosList: PropTypes.array.isRequired
};

export default ListOfVideos;
