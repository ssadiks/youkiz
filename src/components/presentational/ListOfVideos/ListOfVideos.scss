.ListOfVideos {
  display: block;
  padding-bottom: $medium-value;

  &__list {
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    width: 100%;
    margin: auto;

    @include breakpoint (tablet) {
      width: (64 + $double-extra-small-value * 2);
      justify-content: flex-start;
    }

    @include breakpoint (desktop) {
      width: (96 + $double-extra-small-value * 2);
      justify-content: flex-start;
    }
  }

  &__item {
    position: relative;
    flex: 0 0 auto;
    width: 32rem;
    margin-bottom: $double-extra-small-value;
    min-height: 20rem;

    @include breakpoint (tablet) {
      margin-left: $double-extra-small-value;

      &:nth-child(2n+1) {
        margin-left: 0;
      }
    }

    @include breakpoint (desktop) {
      margin-left: $double-extra-small-value;

      &:nth-child(3n+1) {
        margin-left: 0;
      }
    }

    &__more {
      position: absolute !important;
      right: $extra-small-value;
      top:$extra-small-value;
      background-color: $white;
      border-radius: $medium-value;
    }

    &__link {
      display: block;
      height: calc(100% - 3.5rem);
      width: 100%;
      background: grey;
    }

    &__description {
      background-color: $red;
      color: $white;
      padding: $small-value;
    }

    img {
      display: block;
    }
  }
}