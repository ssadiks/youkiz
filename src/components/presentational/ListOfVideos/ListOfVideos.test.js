import React from 'react';
import renderer from 'react-test-renderer';
import ListOfVideos from './ListOfVideos';

jest.mock('material-ui/IconMenu', () => 'IconMenu');
jest.mock('material-ui/MenuItem', () => 'MenuItem');
jest.mock('material-ui/IconButton', () => 'IconButton');

jest.mock('material-ui');

describe('ListOfVideos render', () => {
  const videosList = [{ _id: 1, title: 'video 1' }, { _id: 2, title: 'video 2' }];

  it('if user is authentificated', () => {
    const tree = renderer.create(
      <ListOfVideos
        videosList={videosList}
        userConnected
        editVideo={() => {}}
        handleDeleteVideo={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('if user is not authentificated', () => {
    const tree = renderer.create(
      <ListOfVideos
        videosList={videosList}
        userConnected={false}
        editVideo={() => {}}
        handleDeleteVideo={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
