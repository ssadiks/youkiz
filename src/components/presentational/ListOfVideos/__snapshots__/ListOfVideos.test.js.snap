// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfVideos render if user is authentificated 1`] = `
<div
  className="ListOfVideos"
>
  <h2>
    2 result(s)
  </h2>
  <div
    className="ListOfVideos__list"
  >
    <div
      className="ListOfVideos__item"
    >
      <a
        className="ListOfVideos__item__link"
        href="//www.youtube.com/watch?v=undefined"
        target="_blank"
      >
        <img
          alt=""
          className="media-object"
          src="https://i.ytimg.com/vi/undefined/mqdefault.jpg"
        />
      </a>
      <div
        className="ListOfVideos__item__description"
      >
        <span />
        <span />
        <IconMenu
          anchorOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          className="ListOfVideos__item__more"
          iconButtonElement={
            <IconButton>
              <NavigationMoreVert />
            </IconButton>
          }
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
        >
          <MenuItem
            onClick={[Function]}
            primaryText="Edit"
          />
          <MenuItem
            onClick={[Function]}
            primaryText="Delete"
          />
          <MenuItem
            primaryText="Infos"
          />
        </IconMenu>
      </div>
    </div>
    <div
      className="ListOfVideos__item"
    >
      <a
        className="ListOfVideos__item__link"
        href="//www.youtube.com/watch?v=undefined"
        target="_blank"
      >
        <img
          alt=""
          className="media-object"
          src="https://i.ytimg.com/vi/undefined/mqdefault.jpg"
        />
      </a>
      <div
        className="ListOfVideos__item__description"
      >
        <span />
        <span />
        <IconMenu
          anchorOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
          className="ListOfVideos__item__more"
          iconButtonElement={
            <IconButton>
              <NavigationMoreVert />
            </IconButton>
          }
          targetOrigin={
            Object {
              "horizontal": "left",
              "vertical": "top",
            }
          }
        >
          <MenuItem
            onClick={[Function]}
            primaryText="Edit"
          />
          <MenuItem
            onClick={[Function]}
            primaryText="Delete"
          />
          <MenuItem
            primaryText="Infos"
          />
        </IconMenu>
      </div>
    </div>
  </div>
</div>
`;

exports[`ListOfVideos render if user is not authentificated 1`] = `
<div
  className="ListOfVideos"
>
  <h2>
    2 result(s)
  </h2>
  <div
    className="ListOfVideos__list"
  >
    <div
      className="ListOfVideos__item"
    >
      <a
        className="ListOfVideos__item__link"
        href="//www.youtube.com/watch?v=undefined"
        target="_blank"
      >
        <img
          alt=""
          className="media-object"
          src="https://i.ytimg.com/vi/undefined/mqdefault.jpg"
        />
      </a>
      <div
        className="ListOfVideos__item__description"
      >
        <span />
        <span />
      </div>
    </div>
    <div
      className="ListOfVideos__item"
    >
      <a
        className="ListOfVideos__item__link"
        href="//www.youtube.com/watch?v=undefined"
        target="_blank"
      >
        <img
          alt=""
          className="media-object"
          src="https://i.ytimg.com/vi/undefined/mqdefault.jpg"
        />
      </a>
      <div
        className="ListOfVideos__item__description"
      >
        <span />
        <span />
      </div>
    </div>
  </div>
</div>
`;
